src/vs/workbench/contrib/debug/common/replModel.ts

```diff
@@ -24,10 +116,14 @@ export class ReplModel {
 		return this.replElements;
 	}
 
-	addReplExpression(stackFrame: IStackFrame | undefined, name: string): Promise<void> {
-		const expression = new Expression(name);
-		this.addReplElement(expression);
-		return expression.evaluate(this.session, stackFrame, 'repl');
+	async addReplExpression(stackFrame: IStackFrame | undefined, name: string): Promise<void> {
+		this.addReplElement(new ReplEvaluationInput(name));
+		const result = new ReplEvaluationResult(name);
+		const evaluationResult = await ExpressionContainer.evaluate(name, result, this.session, stackFrame, 'repl');
+		if (evaluationResult) {
+			result.type = evaluationResult.type || result.type;
```

Comment:
Is it possible to move the `type` up to `ExpressionContainer` so we do not have this code duplicated and just put it in the static helper function?